### PR TITLE
prepare trade/single form submit for response

### PR DESCRIPTION
**Pull request by aslink87**
In order to start wiring together a UI to present the response for the form submission in this route, there are a few tasks that needed to be accomplished. This PR focuses on the following:

1. Create necessary types and Zod schema to validate both the form submit object, as well as the form response object on success

2. Update form submit object keys (ie: rename commodityDisplayName to commodityDisplayNames) to match the backend terminology

3. Establish form default values for a few inputs such as 'maxRouteDistance' to simplify the UX

4. update the hook we use to fetch data on form submits `useGetData` to require a response data type to be passed to it

### CODE REVIEW

File: app/_components/trade-routes/single/Form.tsx [hunk 1]
```diff
@@ -381,16 +340,15 @@ const Form: React.FC<FormProps> = ({
                 )
               }
             >
-              <FormLabel>Max Distance From Star</FormLabel>
+              <FormLabel>Max Distance From Star - LS</FormLabel>
```
Comment (chrismou) [inline app/_components/trade-routes/single/Form.tsx]: Can we either stick "In LS" as the placeholder of the input as we do with some other fields, or put (in LS) in brackets (and if you opt for the latter do the same with any other fields where we specify the what unit is required). The - LS looks a bit strange as it is there

image

**Review by chrismou — CHANGES_REQUESTED** [review]
Minor ux niggle otherwise all good 👍🏼

File: app/_components/trade-routes/single/Form.tsx (hunk 1 above)
Comment (aslink87) [reply app/_components/trade-routes/single/Form.tsx]: Establishing default values rules out the first. I'm happy to do the second. My question is, is it necessary at all? Do you think the values in these fields are self-explanatory enough that we can remove the designation entirely?

Comment (chrismou) [reply app/_components/trade-routes/single/Form.tsx]: I don't think having defaults necesarily rules out placeholders, tbh I sort of agree that most of them are intuitive enough, at least to not need it in the label. But using the placeholder is a good fallback for if the input is ever empty for whatever reason and you're not sure what to put in.

That said, does chakra provide any simple solutions for input groups, ie (from bootstrap)

image

Comment (aslink87) [reply app/_components/trade-routes/single/Form.tsx]: How about...

Selection_016

Comment (chrismou) [reply app/_components/trade-routes/single/Form.tsx]: Perfect, that looks really good imo 👍🏼

**Review by chrismou — APPROVED** [review]

**Pull request merged**